adapters/flipp/flipp.go

```diff
@@ -0,0 +1,199 @@
+		PublisherNameIdentifier: &flippExtParams.PublisherNameIdentifier,
+		RequestID:               &request.ID,
+		Height:                  &imp.Banner.Format[0].H,
+		Width:                   &imp.Banner.Format[0].W,
+	}
+
+	var contentCode string
+	if publisherUrl != nil {
+		contentCode = publisherUrl.Query().Get("flipp-content-code")
+	}
+	placement := Placement{
+		DivName: INLINE_DIV_NAME,
+		SiteID:  &flippExtParams.SiteID,
+		AdTypes: getAdTypes(flippExtParams.CreativeType),
+		ZoneIds: flippExtParams.ZoneIds,
+		Count:   &COUNT,
+		Prebid:  &prebidRequest,
+		Properties: &Properties{
+			ContentCode: &contentCode,
+		},
+	}
+
+	var userKey string
+	if request.User != nil && request.User.ID != "" {
+		userKey = request.User.ID
+	} else if flippExtParams.UserKey != "" {
+		userKey = flippExtParams.UserKey
+	} else {
+		uid, err := uuid.NewV4()
+		if err != nil {
+			return nil, fmt.Errorf("Unable to generate user uuid. %v", err)
+		}
+		userKey = uid.String()
+	}
+
+	keywordsArray := strings.Split(request.Site.Keywords, ",")
+	var keywordsSlice []Keyword
+
+	for _, k := range keywordsArray {
+		keywordsSlice = append(keywordsSlice, Keyword(k))
+	}
+
+	campaignRequestBody := CampaignRequestBody{
+		Placements: []*Placement{&placement},
+		URL:        request.Site.Page,
+		Keywords:   keywordsSlice,
+		IP:         request.Device.IP,
+		User: &CampaignRequestBodyUser{
+			Key: &userKey,
+		},
+	}
+
+	adapterReq, err := a.makeRequest(request, campaignRequestBody)
+	if err != nil {
+		return nil, fmt.Errorf("Make request faile with err %v", err)
+	}
+
+	return adapterReq, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var campaignResponseBody CampaignResponseBody
+	if err := json.Unmarshal(responseData.Body, &campaignResponseBody); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = DEFAULT_CURRENCY
+	for _, decision := range campaignResponseBody.Decisions.Inline {
+		b := &adapters.TypedBid{
+			Bid: &openrtb2.Bid{
+				CrID:  fmt.Sprint(decision.CreativeID),
+				Price: *decision.Prebid.Cpm,
+				AdM:   *decision.Prebid.Creative,
+				ID:    fmt.Sprint(decision.AdID),
+				W:     decision.Contents[0].Data.Width,
```

Comment:
here the same question do we need to check out if index?

Reply:
Fixed, added a check before using the `Width` and `Height` values here. If the are not present, nothing is set and that is okay (confirmed in testing)